cpp/src/reconnect_test.cpp

```diff
@@ -129,10 +134,13 @@ class server_connection_handler : public proton::messaging_handler {
 
 class tester_base: public proton::messaging_handler {
   void on_connection_open(proton::connection& c) override {
+    std::string want_url = "amqp://localhost:" + std::to_string(listening_port_);
+
     if (!c.reconnected()) {
       start_count++;
       c.open_sender("messages");
     }
+    ASSERT_EQUAL(c.url(), want_url);
```

Comment:
@astitcher A minor change from our last discussion. CMIIW

Later I realized that c.active() will always be true in on_connection_open(), so no need of adding an assertion for an exception here for the case when the connection is not active.
Hence, here just checking for the URL.